terraform/modules/cancellations-producer-api/cancellations-feed-validator/main.tf

```
@@ -0,0 +1,64 @@
+terraform {
+  required_version = ">= 1.6.6"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.54"
+    }
+  }
+}
+
+module "cancellations_feed_validator" {
+  source = "../../shared/lambda-function"
+
+  environment      = var.environment
+  function_name    = "integrated-data-cancellations-feed-validator"
+  zip_path         = "${path.module}/../../../../src/functions/dist/cancellations-feed-validator.zip"
+  handler          = "index.handler"
+  memory           = 256
+  runtime          = "nodejs20.x"
+  timeout          = 120
+  needs_vpc_access = true
+  custom_sg_id     = var.sg_id
+  subnet_ids       = var.subnet_ids
+  retry_attempts   = 0
+
+  permissions = [
+    {
+      Action   = ["dynamodb:PutItem", "dynamodb:Scan"]
+      Effect   = "Allow",
+      Resource = "arn:aws:dynamodb:${var.aws_region}:${var.aws_account_id}:table/${var.cancellations_subscription_table_name}"
+
+    },
+    {
+      Action = ["ssm:GetParameter"],
+      Effect = "Allow",
+      Resource = [
+        "arn:aws:ssm:${var.aws_region}:${var.aws_account_id}:parameter/subscription/*",
+        "arn:aws:ssm:${var.aws_region}:${var.aws_account_id}:parameter/subscription*"
```

```suggestion
        "arn:aws:ssm:${var.aws_region}:${var.aws_account_id}:parameter/cancellations/subscription/*",
        "arn:aws:ssm:${var.aws_region}:${var.aws_account_id}:parameter/cancellations/subscription*"
```

I had to change the naming for cancellations